contents/tutorials/android-feature-flags.md

```diff
@@ -0,0 +1,238 @@
+            host = POSTHOG_HOST
+        )
+        PostHogAndroid.setup(this, config)
+    }
+}
+```
+
+To get your PostHog API key and host, [sign up to PostHog](https://app.posthog.com/signup). Then, you can find your API key and host in your [project settings](https://app.posthog.com/settings/project).
+
+We now need to register our custom application class. Go to `app/manifests/AndroidManifest.xml` and add `android:name=".MyFeatureFlagsApplication"` within the `<application` tag:
+
+```XML file=app/manifests/AndroidManifest.xml
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools">
+    <!-- ... rest of the config -->
+    <application
+        android:name=".MyFeatureFlagsApplication"
+        <!-- ... rest of the config -->
+    </application>
+</manifest>
+```
+
+To check your setup, build and run your app. Click your button a few times. You should start seeing events in the [activity tab](https://us.posthog.com/events).
+
+<ProductScreenshot
+  imageLight={EventsInPostHogLight} 
+  imageDark={EventsInPostHogDark} 
+  alt="Events captured in PostHog" 
+  classes="rounded"
+/>
+
+## 3. Create a feature flag in PostHog
+
+With PostHog set up, your app is ready for feature flags. To create one, go to the [feature flags tab](https://us.posthog.com/feature_flags) in PostHog and click **New feature flag**. Enter a flag key (like `my-cool-flag`), set the release condition to roll out to 100% of users, and press "Save."
+
+<ProductScreenshot
+  imageLight={CreateFlagLight} 
+  imageDark={CreateFlagDark} 
+  alt="Feature flag created in PostHog" 
+  classes="rounded"
+/>
+
+You can customize your [release conditions](/docs/feature-flags/creating-feature-flags#release-conditions) with rollout percentages, and [user](/docs/product-analytics/user-properties) or [group properties](/docs/product-analytics/group-analytics) to fit your needs.
+
+## 4. Implement the flag code
+
+To implement the feature flag, we: 
+
+1. Fetch the `my-cool-flag` flag using [`PostHog.isFeatureEnabled()`](/docs/libraries/android#feature-flags).
+2. Change the background color of `SecondScreen` based on the value of the flag.
+
+To do this, update the code in `MainActivity.kt` with the following:
+
+```kt file=MainActivity.kt
+import androidx.compose.runtime.LaunchedEffect
+// ... [Rest of your imports and MainActivity code]
+
+
+@Composable
+fun SecondScreen() {
+    val context = LocalContext.current
+    val isTestVariant = remember { mutableStateOf(false) }
+
+    // LaunchedEffect to call PostHog.getFeatureFlag() once when the composable is initialized
+    LaunchedEffect(key1 = Unit) {
+        isFlagEnabled.value = PostHogAndroid.with(context, PostHogAndroidConfig(
+            apiKey = "phc_Is2364QKW0QKHbXWVtgh8Yugx8xAvdfL71JmBaKD8mG",
+            host = "https://app.posthog.com"
+        )).isFeatureEnabled("my-cool-flag")
+    }
```

Comment:
@marandaneto  is this correct? Feels strange to have to pass through the API key and host again, even though we already did it in the app subclass

Reply:
https://posthog.com/docs/libraries/android
`PostHog.isFeatureEnabled("flag-key")`

Reply:
Hmm If I do that I get a unresolved reference. I've added these imports to the file:
```
import com.posthog.android.PostHogAndroid
import com.posthog.android.PostHogAndroidConfig
```

<img width="678" alt="Screenshot 2024-02-15 at 12 30 36 PM" src="https://github.com/PostHog/posthog.com/assets/7090054/ec3477f5-80e2-45e3-b3a7-8f1b87117025">

Reply:
just import the class as the IDE suggests.

`import com.posthog.PostHog`

Reply:
https://github.com/PostHog/posthog.com/pull/7829